src/obsolete-reporter.js

```diff
@@ -0,0 +1,64 @@
+const fs = require("fs");
+const path = require("path");
+
+const TOUCHED_FILE_LIST_PATH = path.join(
+  process.cwd(),
+  ".jest-image-snapshot-touched-files"
+);
+
+const IS_ENABLED = !!process.env.JEST_IMAGE_SNAPSHOT_TRACK_OBSOLETE;
```

Comment:
gated behind a ENV flag

Reply:
I’d like this flag to be opt-out, rather than opt-in. I think the majority of users will likely want to have it on by default. Just a thought.

Reply:
While I agree the majority of users will want to have it on, it requires explicit usage of a custom reporter configured in jest that will be fundamentally optional. the reporter might be able to flip it on by default once it's included but I haven't looked into that yet (not sure if jest will respect whatever reporters do to `process.env` when spawning the workers)

Reply:
@patrickhulce Totally understood. I'm not very familiar with how Jest's test reporter system works, but looking at this code, it appears that nothing in this file will ever even be executed unless the user adds `jest-image-snapshot/src/obsolete-reporter.js` to their Jest configuration, correct? In which case it feels like the environment variable is redundant.

Reply:
> nothing in this file will ever even be executed unless the user adds jest-image-snapshot/src/obsolete-reporter.js to their Jest configuration, correct?

Correct.

> I probably wouldn't add the environment variable gate at all if it were me, but I'd settle for using it as an opt-out escape hatch.

AFAIK, it is intentionally impossible to tell within each jest worker at the level of `toMatchImageSnapshot()` what reporter is being used. Because of that, we need *some* way of signaling to the worker processes that it needs to write to this "touched files" list. `process.env` is a straightforward, safe option.

Alternatives I see:

- Keep this environmental variable internal only and enable it by default (already mentioned, not sure if jest will respect whatever reporters do to process.env when spawning the workers)
- Treat the very existence of the magic file name as a signal to write to it. Prone to cleanup issues, but I'd favor this approach if we decided to make this enabled by default.

Reply:
I think we can start with opt out and then if it works well and we can figure out how to have it cleanly enabled by default then we can make it a default. Easier to add than to remove features ;)